scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,417 @@
+        
+        public UserConfig (
+            bool profanityFilterRemoveEnabled,
+            bool profanityFilterMaskEnabled,
+            string[]? languageIDLanguages,
+            string? inputFile,
+            string? outputFile,
+            string? phraseList,
+            bool suppressOutputEnabled,
+            bool partialResultsEnabled,
+            string? stablePartialResultThreshold,
+            bool srtEnabled,
+            bool trueTextEnabled,
+            string subscriptionKey,
+            string region
+            )
+        {
+            this.profanityFilterRemoveEnabled = profanityFilterRemoveEnabled;
+            this.profanityFilterMaskEnabled = profanityFilterMaskEnabled;
+            this.languageIDLanguages = languageIDLanguages;
+            this.inputFile = inputFile;
+            this.outputFile = outputFile;
+            this.phraseList = phraseList;
+            this.suppressOutputEnabled = suppressOutputEnabled;
+            this.partialResultsEnabled = partialResultsEnabled;
+            this.stablePartialResultThreshold = stablePartialResultThreshold;
+            this.srtEnabled = srtEnabled;
+            this.trueTextEnabled = trueTextEnabled;
+            this.subscriptionKey = subscriptionKey;
+            this.region = region;
+        }
+    }
+
+    public class TooFewArguments : Exception
+    {
+        public TooFewArguments() : base() { }
+    }
+
+    class Program
+    {
+
+/*
+Helper functions
+*/
+
+        static string? GetCmdOption(string[] args, string option)
+        {
+            int index = Array.IndexOf(args, option);
+            if (index > -1 && index < args.Length - 2)
+            {
+// We found the option (for example, "-o"), so advance from that to the value (for example, "filename").
+                return args[index + 1];
+            }
+            else {
+                return null;
+            }
+        }
+
+        static bool CmdOptionExists(string[] args, string option)
+        {
+            return args.Contains (option);
+        }
+
+        static string V2EndpointFromRegion(string region)
+        {
+// Note: Continuous language identification is supported only with v2 endpoints.
+            return "wss://" + region + ".stt.speech.microsoft.com/speech/universal/v2";
+        }
+
+        static string TimestampFromSpeechRecognitionResult(SpeechRecognitionResult result, UserConfig userConfig)
+        {
+            DateTime startTime = new DateTime(result.OffsetInTicks);
+            DateTime endTime = startTime.Add(result.Duration);
+            string format;
+
+            if(userConfig.srtEnabled)
+            {
+// SRT format requires ',' as decimal separator rather than '.'.
+                format = "HH:mm:ss,fff";
+            }
+            else
+            {
+                format = "HH:mm:ss.fff";
+            }
+
+            return String.Format("{0} --> {1}", startTime.ToString(format), endTime.ToString(format));
```

Comment:
It's a tradeoff since you can't both assign the format as a variable and use it inside of interpolated strings, but it might be worth using string interpolation for the cleanliness, e.g. remove the above and instead:
```
// SRT caption format requires ',' as the decimal separator instead of '.'
return userConfig.srtEnabled
    ? $"{startTime:HH:mm:ss,fff} --> {endTime:HH:mm:ss,fff}"
    : $"{startTime:HH:mm:ss.fff} --> {endTime:HH:mm:ss.fff}";
```
(My feelings mixed and not as strong on this one since it's not perfect either way)